packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -283,7 +283,7 @@ export class Alarm extends AlarmBase {
                 stat: stat.statistic,
                 unit: stat.unitFilter,
               },
-              id: stat.metricName,
+              id: id,
```

Comment:
```suggestion
              id,
```

Otherwise the linter will yell at you

Reply:
My bad. Thanks for pointing me out :)